contribs/github-bot/client/client.go

```diff
@@ -0,0 +1,248 @@
+			comment,
+		)
+		if err != nil {
+			gh.Logger.Errorf("Unable to edit bot comment with ID %d: %v", comment.GetID(), err)
+			return nil
+		}
+		return editComment
+	}
+}
+
+func (gh *GitHub) ListTeamMembers(team string) []*github.User {
+	var (
+		allMembers []*github.User
+		opts       = &github.TeamListTeamMembersOptions{
+			ListOptions: github.ListOptions{
+				PerPage: PageSize,
+			},
+		}
+	)
+
+	for {
+		members, response, err := gh.Client.Teams.ListTeamMembersBySlug(
+			gh.Ctx,
+			gh.Owner,
+			team,
+			opts,
+		)
+		if err != nil {
+			gh.Logger.Errorf("Unable to list members for team %s: %v", team, err)
+			return nil
+		}
+
+		allMembers = append(allMembers, members...)
+
+		if response.NextPage == 0 {
+			break
+		}
+		opts.Page = response.NextPage
+	}
+
+	return allMembers
+}
+
+func (gh *GitHub) IsUserInTeams(user string, teams []string) bool {
+	for _, team := range teams {
+		for _, member := range gh.ListTeamMembers(team) {
+			if member.GetLogin() == user {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+func (gh *GitHub) ListPrReviewers(prNum int) *github.Reviewers {
+	var (
+		allReviewers = &github.Reviewers{}
+		opts         = &github.ListOptions{
+			PerPage: PageSize,
+		}
+	)
+
+	for {
+		reviewers, response, err := gh.Client.PullRequests.ListReviewers(
+			gh.Ctx,
+			gh.Owner,
+			gh.Repo,
+			prNum,
+			opts,
+		)
+		if err != nil {
+			gh.Logger.Errorf("Unable to list reviewers for PR %d: %v", prNum, err)
+			return nil
+		}
+
+		allReviewers.Teams = append(allReviewers.Teams, reviewers.Teams...)
+		allReviewers.Users = append(allReviewers.Users, reviewers.Users...)
+
+		if response.NextPage == 0 {
+			break
+		}
+		opts.Page = response.NextPage
+	}
+
+	return allReviewers
+}
+
+func (gh *GitHub) ListPrReviews(prNum int) []*github.PullRequestReview {
+	var (
+		allReviews []*github.PullRequestReview
+		opts       = &github.ListOptions{
+			PerPage: PageSize,
+		}
+	)
+
+	for {
+		reviews, response, err := gh.Client.PullRequests.ListReviews(
+			gh.Ctx,
+			gh.Owner,
+			gh.Repo,
+			prNum,
+			opts,
+		)
+		if err != nil {
+			gh.Logger.Errorf("Unable to list reviews for PR %d: %v", prNum, err)
+			return nil
+		}
+
+		allReviews = append(allReviews, reviews...)
+
+		if response.NextPage == 0 {
+			break
+		}
+		opts.Page = response.NextPage
+	}
+
+	return allReviews
+}
+
+func New(params param.Params) *GitHub {
+	gh := &GitHub{
+		Owner:  params.Owner,
+		Repo:   params.Repo,
+		DryRun: params.DryRun,
+	}
+
+	// Detect if the current process was launched by a GitHub Action and return
+	// a logger suitable for terminal output or the GitHub Actions web interface
+	gh.Logger = logger.NewLogger(params.Verbose)
+
+	// Create context with timeout if specified in the parameters
+	if params.Timeout > 0 {
+		//nolint:govet,lostcancel // Keeping this cancel function is useless in this context
+		gh.Ctx, _ = context.WithTimeout(context.Background(), time.Duration(params.Timeout)*time.Millisecond)
```

Comment:
Even though I understand the value of the govet check and the best practice related to it, in this specific case, I'm not entirely convinced that it's necessary since the client package isn't meant to be generic/reusable, and deferring a cancel in the main function as it stands right now seems useless to me. 

But I don't mind applying your suggestion. Done : https://github.com/gnolang/gno/pull/3037/commits/8d62467b5a04747158f0f625fa009f3820e20e28